tools/clang-format-check.sh

```diff
@@ -1,6 +1,9 @@
 #!/bin/bash
 
-clang-format-5.0 -style=file -output-replacements-xml $(git ls-files | grep '\.cc\|\.h$') | grep -c "<replacement "
+set -e
```

Comment:
Because of the missing `-e` and we're piping this would exit 0 even if it failed such as command not failed. Which clang-format-5.0 never existed so this silently passes.

As you can see, master isn't even running these and silently passing:

<img width="1122" alt="screen shot 2018-06-19 at 13 41 48" src="https://user-images.githubusercontent.com/44164/41597878-99191ffa-73c6-11e8-9900-b67417fd90f5.png">

The clang-format tests are failing and that's because the code doesn't follow the rules. It's outside of this PR to fix so as the lint is not "required" we can continue and merge PR and fix files not confirming to clang-format separately.

Reply:
Fixing this will likely conflict with #552 so we should likely wait until thats merged.